File: bindings/go/tests/test_extractor_test.go
package tests

import (
	"context"
	"testing"

	tiktokdirect "github.com/codewithwan/tiktok-direct-go"
)

const URL = "https://vt.tiktok.com/ZSxvYRvoR/"

func TestInvalidUrlRaisesError(t *testing.T) {
	extractor := tiktokdirect.New()
	_, err := extractor.Extract(context.Background(), "invalid-url")
	if err == nil {
		t.Error("Expected error for invalid url, got nil")
	}

	_, err = tiktokdirect.Extract(context.Background(), "invalid-url")
	if err == nil {
		t.Error("Expected error for invalid url in package function, got nil")
	}
}

func TestExtractRealUrl(t *testing.T) {
	extractor := tiktokdirect.New()
	meta, err := extractor.Extract(context.Background(), URL)
	if err != nil {
		t.Fatalf("Failed to extract real URL: %v", err)
	}

	if meta["quality"] != "complete" && meta["quality"] != "partial" {
		t.Errorf("Expected complete or partial, got %v", meta["quality"])
	}
	if fmtID := meta["video_id"]; fmtID == nil || fmtID == "" {
		t.Error("Expected video_id, got empty")
	}
	if fmtTitle := meta["title"]; fmtTitle == nil || fmtTitle == "" {
		t.Error("Expected title, got empty")
	}
}

File: bindings/node/tests/test_extractor.test.js
import assert from "node:assert/strict";
import { test } from "node:test";
import { TikTokExtractor, extract, download } from "../src/index.js";

const URL = "https://vt.tiktok.com/ZSxvYRvoR/";

test("test_module_exports", () => {
  assert.ok(TikTokExtractor);
  assert.ok(extract);
  assert.ok(download);
});

test("test_invalid_url_raises_error", async () => {
  const extractor = new TikTokExtractor();
  await assert.rejects(
    async () => await extractor.extract("invalid-url")
  );

  await assert.rejects(
    async () => await extract("invalid-url")
  );
});

test("test_extract_real_url", async () => {
  const extractor = new TikTokExtractor();
  const meta = await extractor.extract(URL);

  assert.equal(meta.quality, "complete");
  assert.ok(meta.video_id);
  assert.ok(meta.title);
});
